nixos/modules/services/web-apps/writefreely.nix

```diff
@@ -0,0 +1,102 @@
+{config, lib, pkgs, ...}:
+
+with lib;
+
+let
+  cfg = config.services.writefreely;
+  configFile = pkgs.writeText "config.ini" (generators.toINI {} cfg.settings);
+  settingsFormat = pkgs.formats.json {};
+in {
+  options.services.writefreely = {
+    enable = mkEnableOption "WriteFreely, a clean, Markdown-based publishing platform made for writers. Write together, and build a community.";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.writefreely;
+      description = ''
+        Overridable attribute of the WriteFreely package to use.
+      '';
+    };
+
+    user = mkOption {
+      type = types.str;
+      default = "writefreely";
+      description = "User account under which writefreely runs.";
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "writefreely";
+      description = "User account under which writefreely runs.";
+    };
+
+    dataDir = mkOption {
+      default = "/var/lib/writefreely";
+      type = types.str;
+      description = "WriteFreely data directory.";
+    };
+
+    settings = lib.mkOption {
```

Comment:
It would be great if there is a plan to support a `secretFile` that can be loaded into `LoadCredential` and merge with the final INI file to avoid secrets in the Nix store, but that's okay if it's not in your scope.